tools/file_packager.py

```diff
@@ -914,6 +914,10 @@ def generate_js(data_target, data_files, metadata):
           for (var chunkId = 0; chunkId < chunkCount; chunkId++) {
             var getRequest = packages.get(`package/${packageName}/${chunkId}`);
             getRequest.onsuccess = function(event) {
+              if (!event.target.result) {
```

Comment:
Why would the `onsuccess` callback be triggered in this case?

Reply:
I'm not sure why it's implemented that way in IDBObjectStore, but that seems to be how it works.

Perhaps a hint is in:
https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/get
where it says:
> This method produces the same result for: a) a record that doesn't exist in the database and b) a record that has an undefined value.

Of note: The earlier implementation in `checkCachedPackage()` does already handle this case:
https://github.com/emscripten-core/emscripten/blob/4b9a5f77efab12859152ac71e7f695890f2832ab/tools/file_packager.py#L892-L895

The handling is just missing in `fetchCachedPackage()`, which this PR adds.

Reply:
I see, so when a record is not found that is not an `error` but a `success` with an empty result?

Reply:
> I see, so when a record is not found that is not an `error` but a `success` with an empty result?

Yep, seems to be